hyperspy/io.py

```diff
@@ -178,24 +178,42 @@ def load(filenames=None,
         if len(filenames) > 1:
             messages.information('Loading individual files')
         if stack is True:
-            signal = []
             for i, filename in enumerate(filenames):
                 obj = load_single_file(filename,
                                        **kwds)
-                signal.append(obj)
-            signal = hyperspy.utils.stack(signal,
-                                          axis=stack_axis,
-                                          new_axis_name=new_axis_name,
-                                          mmap=mmap, mmap_dir=mmap_dir)
-            signal.metadata.General.title = \
-                os.path.split(
+                if i == 0:
+                    if isinstance(obj, (list, tuple)):
+                        n = len(obj)
+                    else:
+                        n = 1
+                    signals = [[] for j in xrange(n)]
+                else:
+                    if isinstance(obj, (list, tuple)):
```

Comment:
What's the point of the next four lines? If the code above is correct the ValueError exception will never be raised, and if it is incorrect the exception will confuse the users as it is not their fault that it doesn't work...